### PR TITLE
Test user_signups_approved_with_method_total metric

### PR DESCRIPTION
**Pull request by rajivnathan**
Paired with https://github.com/codeready-toolchain/host-operator/pull/804

### CODE REVIEW

File: test/metrics/metrics_test.go [hunk 1]
```diff
@@ -19,6 +19,72 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// TestMetricsWhenUsersAutomaticallyApproved verifies that `UserSignupsApprovedMetric` and `UserSignupsApprovedWithMethodMetric` counters are increased when users are approved
+func TestMetricsWhenUsersAutomaticallyApproved(t *testing.T) {
```
Comment (alexeykazakov) [inline test/metrics/metrics_test.go:23]: Shouldn't we also rename `TestMetricsWhenUsersDeactivated()` to `TestMetricsWhenUsersManuallyApproved()`? Those two functions look identical and test both activation and deactivation but the names are a bit confusing...

**Review by alexeykazakov — APPROVED** [review]

File: test/metrics/metrics_test.go (hunk 1 above)
Comment (rajivnathan) [reply test/metrics/metrics_test.go:23]: Ok, I renamed them to `TestMetricsWhenUsersManuallyApprovedAndThenDeactivated` and `TestMetricsWhenUsersAutomaticallyApprovedAndThenDeactivated`

**Review by mfrancisc — APPROVED** [review]
Looks fine 👍 

I don't think it's super important but was wondering - should we test `Ban` and `Disable` with a mix of `manual` and `automatic` approved usersignups ? Now I see the `TestMetricsWhenUsersBanned` and `TestMetricsWhenUserDisabled` are being provisioned "only" with manual approved users.

**Comment by rajivnathan** [comment]
>I don't think it's super important but was wondering - should we test Ban and Disable with a mix of manual and automatic approved usersignups ? Now I see the TestMetricsWhenUsersBanned and TestMetricsWhenUserDisabled are being provisioned "only" with manual approved users.

TestMetricsWhenUsersBanned and TestMetricsWhenUserDisabled aren't impacted by the approval type so it's not really necessary.

**Pull request merged**